src/app/components/StakingDateSelector/index.tsx

```diff
@@ -44,21 +44,24 @@ export function StakingDateSelector(props: Props) {
   const [selectedMonth, setSelectedMonth] = useState('');
   const [selectedDay, setSelectedDay] = useState('');
 
-  const dateWithoutStake = filteredDates.reduce((unique: any, o: any) => {
-    let isFound = itemDisabled.some((b: { key: any }) => {
-      return b.key === o.key;
-    });
-    if (!isFound) unique.push(o);
-    return unique;
-  }, []);
+  const [dateWithoutStake, avaliableYears, avaliableMonth] = useMemo(() => {
+    const dateWithoutStake = filteredDates.reduce((unique: any, o: any) => {
```

Comment:
I don't like this, use correct typing. `unique` is `DateItem[]` and `o` is `DateItem`. 

Also, adjust the naming - change `unique` to `uniqueDates` and `o` to `item` (or something else if you find it more suitable).